src/main/java/io/jawg/osmcontributor/ui/activities/PhotoActivity.java

```diff
@@ -217,14 +217,21 @@ protected void onCreate(Bundle savedInstanceState) {
         // Init parameters.
         latitude = getIntent().getDoubleExtra("latitude", 0);
         longitude = getIntent().getDoubleExtra("longitude", 0);
-        imageAdapters = new HashMap<Long, ImageAdapter>();
 
         // Init listener and view
         initScrollListener();
         initOnClickItemListener();
         initView();
     }
 
+    @Override
```

Comment:
I just double checked and there are a few leaks in this activity.

When creating the activity, registration is performed on eventbus, and not unregistered when destroyed. A context is therefore being leaked (aaaaaaaaaaaahhh :O)

Same thing when pausing the activity (screen turned off or switched app), Asynctask still runs and throws event. App should crash as Activity is not shown on screen when results are available.

Suggestion:
register in onresume, unregister in onpause, this solves problem 1.
for problem 2, two possibilities:
1) you want to keep on downloading the images in background even though the app is not on screen. Problem is, the events will be thrown and never caught with the subscribers (because remember, you unsubscribed the activity from the eventbus). 
So you need to launch task in onresume (after registration on eventbus) only if not launched before (one-shot).
Then, you can use the post sticky option from eventbus from the asynctask itself (see docs). This will allow the event to stay in-memory until the re-registered activity can consume it.

2) you would rather be safe, even if it means downloading again when switching back. Then you need to:
Launch the new task in onresume, and cancel it always in onPause.

I haven't taken a deeper look into the rest, but there might be other bugs and leaks related to the activity lifecycle.
I would suggest to use Android LeakCanary (a leak detection library) to track your contexts and eventually detect leaks.
Don't hesitate to also use monitoring tools to check your memory profile and footprint. Posting image results is easy but might be heavy on memory. Solution 2 would then be a healthier choice as sticky events can potentially stay in-memory for a long time.